pytorch_lightning/utilities/parsing.py

```diff
@@ -225,12 +226,23 @@ def save_hyperparameters(
         for local_args in collect_init_args(frame, []):
             init_args.update(local_args)
 
-    if ignore is not None:
-        if isinstance(ignore, str):
-            ignore = [ignore]
-        if isinstance(ignore, (list, tuple)):
-            ignore = [arg for arg in ignore if isinstance(arg, str)]
-        init_args = {k: v for k, v in init_args.items() if k not in ignore}
+    if ignore is None:
+        ignore = []
+    elif isinstance(ignore, str):
+        ignore = [ignore]
+    elif isinstance(ignore, (list, tuple)):
+        ignore = [arg for arg in ignore if isinstance(arg, str)]
+
+    for k in list(init_args):
+        if k in ignore:
+            del init_args[k]
+            continue
+
+        if isinstance(init_args[k], nn.Module):
+            rank_zero_warn(
+                f"Attribute {k!r} is an instance of `nn.Module` and is already saved during checkpointing."
+                " It is recommended to ignore them using `self.save_hyperparameters(ignore=[k!r])`."
+            )
```

Comment:
I'd prefer

```python
modules = [arg for arg in init_args if arg not in ignore and isinstance(arg, nn.Module)]
for module in modules:
    warn(...)
```

Reply:
I am doing 2 things here. Filtering the ones that are required to be ignored and raising a warning. can't seem to find a way to do both in a single loop.

Reply:
But what's the issue with doing it in 2 loops as in my suggestion?

Reply:
> But what's the issue with doing it in 2 loops as in my suggestion?

Probably, `del init_args[k]` is also required (and hence not a list comprehension)?